apps/dashboard/src/app/label-sets/label-set/label-set.component.ts

```diff
@@ -123,91 +133,61 @@ export class LabelSetComponent implements OnDestroy {
       return;
     }
-    this.ontology?.modifyLabel(title, changes);
+    this.labelSet?.modifyLabel(title, { title: newTitle });
     this.labelOrder = this.getLabelOrder();
-    this.hasChanges = true;
+    this.hasChanges = this.hasChanged();
     this.cdr.markForCheck();
   }
 
   deleteLabel(title: string) {
-    this.ontology?.deleteLabel(title);
+    this.labelSet?.deleteLabel(title);
     this.labelOrder = this.getLabelOrder();
-    this.hasChanges = true;
+    this.hasChanges = this.hasChanged();
     this.cdr.markForCheck();
   }
 
   drop(event: CdkDragDrop<string[]>) {
     moveItemInArray(this.labelOrder, event.previousIndex, event.currentIndex);
-    this.ontology?.setLabelOrder(this.labelOrder);
-    this.hasChanges = true;
+    this.labelSet?.setLabelOrder(this.labelOrder);
+    this.hasChanges = this.hasChanged();
     this.cdr.markForCheck();
   }
 
-  saveOntology(): void {
-    if (!this.ontologyForm.valid) return;
+  saveLabelSet(): void {
+    if (!this.labelSetForm.valid || !this.labelSet) {
+      return;
+    }
+
     let slug: string;
     if (this.addNew) {
-      const slugs = Object.keys(this.initialLabels!);
-      slug = STFUtils.generateUniqueSlug(this.ontologyForm.value.title, slugs);
+      const slugs = Object.keys(this.labelSetBackup!);
+      slug = STFUtils.generateUniqueSlug(this.labelSetForm.getRawValue().title, slugs);
     } else {
-      slug = this.ontologySlug!;
+      slug = this.labelSetId as string;
     }
+
     this.labelsService
-      .saveLabelSet(slug, this.ontology!.getCopy())
+      .saveLabelSet(slug, this.labelSet.getCopy())
       .pipe(
         switchMap(() => this.labelsService.refreshLabelsSets()),
-        takeUntil(this.unsubscribeAll),
-      )
-      .subscribe(() => {
-        this.goToOntologyList();
-      });
-  }
-
-  deleteOntology(): void {
-    this.modalService
-      .openConfirm({
-        title: 'label-set.delete_confirm_title',
-        description: 'label-set.delete_warning_extra',
-        isDestructive: true,
-      })
-      .onClose.pipe(
-        filter((confirm) => !!confirm),
-        switchMap(() => this.labelsService.deleteLabelSet(this.ontologySlug!)),
-        switchMap(() => this.labelsService.refreshLabelsSets()),
-        takeUntil(this.unsubscribeAll),
+        take(1),
       )
-      .subscribe(() => {
-        this.goToOntologyList();
-      });
-  }
-
-  cancelChanges(): void {
-    if (this.initialLabels) {
-      this.initState(this.initialLabels);
-    }
+      .subscribe(() => this.goToLabelSetList());
   }
 
   isDuplicatedLabel(title: string): boolean {
-    return !!this.ontology?.labels?.find((label) => label.title === title);
-  }
-
-  setMultiple(selected: boolean) {
-    if (this.ontology) {
-      this.ontology.multiple = selected;
-      this.hasChanges = true;
-      this.cdr.markForCheck();
-    }
+    return !!this.labelSet?.labels?.find((label) => label.title === title);
   }
 
   showDuplicationWarning(title: string) {
```

Comment:
A very small thing, text`label-set.eduplicated_label` is misssing

Reply:
good catch, thanks!!